ui/app/clinical/consultation/controllers/diagnosisController.js

```diff
@@ -265,7 +265,7 @@ angular.module('bahmni.clinical')
             $scope.checkInvalidDiagnoses = function () {
                 $scope.errorMessage = "";
                 $scope.consultation.newlyAddedDiagnoses.forEach(function (diagnosis) {
-                    if (isInvalidDiagnosis(diagnosis)) {
+                    if (isInvalidDiagnosis(diagnosis) || isDuplicateDiagnosis(diagnosis)) {
```

Comment:
The error message to be shown in case of inValidDiagnosis() and duplicateDiagnosis() is the same?

Reply:
Yes @gsluthra. In both cases, we are checking for duplicate entries. isInvalidDiagnosis() checks for duplicates in newlyAddedDiagnoses whereas isDuplicateDiagnosis() compares newlyAddedDiagnoses with savedDiagnosesFromCurrentEncounter.

Reply:
Then should we merge the two methods into one, so that here, it only reads it as "isDuplicateDiagnosis()"? Or rename the isInvalidDiagnosis() to: isDuplicateWithNewlyAddedDiagnosis(), and the second method to isDuplicateWithSavedDiagnosis()?

Reply:
@gsluthra We have renamed the methods, since merging these two functions are causing some inconsistencies.